hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/RootEntityHandler.java

```diff
@@ -48,26 +53,44 @@ public RootEntityHandler(
   @Override
   public NamespaceSummaryResponse getSummaryResponse()
           throws IOException {
-    NamespaceSummaryResponse namespaceSummaryResponse =
-            new NamespaceSummaryResponse(EntityType.ROOT);
+
     List<OmVolumeArgs> volumes = listVolumes();
-    namespaceSummaryResponse.setNumVolume(volumes.size());
     List<OmBucketInfo> allBuckets = listBucketsUnderVolume(null);
-    namespaceSummaryResponse.setNumBucket(allBuckets.size());
     int totalNumDir = 0;
     long totalNumKey = 0L;
     for (OmBucketInfo bucket : allBuckets) {
       long bucketObjectId = bucket.getObjectID();
       totalNumDir += getTotalDirCount(bucketObjectId);
       totalNumKey += getTotalKeyCount(bucketObjectId);
     }
+    CountStats countStats = new CountStats(
+        volumes.size(), allBuckets.size(), totalNumDir, totalNumKey);
 
-    namespaceSummaryResponse.setNumTotalDir(totalNumDir);
-    namespaceSummaryResponse.setNumTotalKey(totalNumKey);
+    NamespaceSummaryResponse namespaceSummaryResponse =
+        NamespaceSummaryResponse.newBuilder()
+            .setEntityType(EntityType.ROOT)
+            .setCountStats(countStats)
+            .setObjectDBInfo(getPrefixObjDbInfo())
+            .setStatus(ResponseStatus.OK)
+            .build();
 
     return namespaceSummaryResponse;
   }
 
+  private ObjectDBInfo getPrefixObjDbInfo()
+      throws IOException {
+    OmPrefixInfo omPrefixInfo =
+        getOmMetadataManager().getPrefixTable().
+            getSkipCache(OzoneConsts.OM_KEY_PREFIX);
+    ObjectDBInfo objectDBInfo = new ObjectDBInfo();
```

Comment:
Same as `BucketObjectDBInfo`, create constructor or builder which takes `OmPrefixInfo` and create `ObjectDBInfo` object.